packages/@glimmer/runtime/lib/syntax/functions.ts

```diff
@@ -191,6 +191,13 @@ export class InvokeDynamicLayout implements DynamicInvoker<ProgramSymbolTable> {
       if (hasEval) lookup![callerNames[i]] = value;
     }
 
+    let numPositionalArgs = stack.pop<number>();
+
+    assert(typeof numPositionalArgs === 'number', 'zomg');
```

Comment:
We could start sprinkling these in too, but we probably want a clearer message like [BUG] Missing argc: ...